### ISSUE TITLE
Framebuffer problem for macbook with gt750m

### ISSUE THREAD

**Issue by liangtb** [yifanlu/NVIDIAEGPUSupport#1]
Hi, I have macbook pro retina 15 2013-late with gt750m, This model can't work with eGPU either AMD_based or Nvidia_based. 

Because of the nvidiaframebuffer this problem can not be solved, but I see your solution, I think maybe this will work. How I can modify it for AMD_eGPU, just solved the framebuffer problem.

**Comment by liangtb** [comment]
when I remove Geforce related driver on 10.13beta and under, macbook with gt750 will boot in iGPU mode, this make AMD_eGPU work perfectly, but disabling gt750m makes the screen light adjustment  non-functional, in addition, even though gt750 is disabled, but the power consumption of gt750 is still here, unlike the auto-switch gpu will no consumption for gt750m on iGPU mode.

**Comment by yifanlu** [comment]
Sorry I'm not familiar with what the "nvidia framebuffer" problem is?

**Comment by liangtb** [comment]
Apple introduced AMD-eGPU with High Sierra, this worked on all iGPU-based and AMD-based macbooks or mac devices, but not worked on NVIDIA-based macbooks, such as macbook pro retina 15 2013-late with gt750m.
When remove NVIDIA and Gefore driver, enable it to iGPU-based macbook, it worked with AMD_eGPU, so this is the problem only between NVIDIA-based graphics and AMD_eGPU, when plug AMD_eGPU to NVIDI-based macbook, it got black-screen both internal and external screens, so there are two options, either disable NVIDIA graphics on boot or solve the compatibility between NVIDIA-based graphics and AMD_eGPU.
First option is confirmed wrong, because when disable the NVIDIA graphics on boot, the function of screen light-adjustment did not work.
So there's one option left, solve the  compatibility between NVIDIA-based graphics and AMD_eGPU.
I see your solution bypass the check, so if when boot or even log out with hot-plug, can disable the framebuffer check, and make AMD_eGPU configure firstly, then it will work fine with AMD-eGPU.
look forward to receiving you reply, thanks!

**Comment by liangtb** [comment]
I tested your script on macbook pro retina 15 2013-late with gt750m on high sierra. It worked good.

**Comment by liangtb** [comment]
It happens on all NVIDIA-based macbook such as gt750m and gt650m. I appreciate if you can solve this, I can help test it on anytime.

**Comment by liangtb** [comment]
yes, under sierra and above it, it worked with goalque's script. As you said, it's same as sierra, so I tested my gtx970 and rx580, gtx970 worked fine, but rx580 is same as before, incompatible with high sierra.

**Comment by liangtb** [comment]
The RX580 I used is sapphire which Apple officially chose in develop kit. On iGPU boot, it worked fine with high sierra, but as I mentioned above, there are several problems existed, such as light-adjustment and extra power consumption, which related NVDA and Geforce controlled.
So should with NVDA and Geforce kext, but fix it on boot or log process, this will work perfectly.

**Comment by yifanlu** [comment]
Interesting. I'm not sure I can fix it though as I don't have anything to test with and don't really understand the problem.

**Comment by liangtb** [comment]
I can test it, if you have some idea, somebody adopted remove Geforce kext on high sierra beta4, it's okay, but above high sierra beta4 even high sierra official, it not works.

**Comment by liangtb** [comment]
If you can delay NVIDIA and Geforce driver until AMD_eGPU loaded, at least can successfully boot even though with the hot-plug function.

**Comment by liangtb** [comment]
If you can try to solve it, I appreciate it. Thanks!
I can help test it anytime. My mail is ltingbiao@gmail.com.

That's all, Thank you again!

**Comment by liangtb** [comment]
I have extract my sysdiagnose when eGPU hot plug in, Do you need this? I will send to you.

**Comment by liangtb** [comment]
https://drive.google.com/a/pcc.edu/file/d/0BwzI2XztZI6KNmhPOWN0ZEZFbDQ/view?usp=sharing

**Comment by liangtb** [comment]
https://drive.google.com/file/d/0BwzI2XztZI6KNmhPOWN0ZEZFbDQ/view?usp=sharing

**Comment by yifanlu** [comment]
I removed your link since the dump contains a lot of identifying information. I noticed that all the AMD drivers are loaded properly and there isn't any error.

What you observe is that both screens go blank?

**Comment by liangtb** [comment]
yes, all go black, I guess gt750m block the thunderblot function or conflict with AMD-eGPU, I can load AMD-eGPU on safe boot or recovery mode, which both have not NVIDIA acceleration.

**Comment by liangtb** [comment]
[safe.txt](https://github.com/yifanlu/NVIDIAEGPUSupport/files/1419359/safe.txt)
This is safe boot and normal boot loaded kext list, as follows.
[normal.txt](https://github.com/yifanlu/NVIDIAEGPUSupport/files/1419358/normal.txt)

**Comment by yifanlu** [comment]
I don't think there's an easy fix. Everything is loaded correctly but there's likely some bug in apple's rendering code that causes these issues.

**Comment by mayankk2308** [comment]
That's probably the case. One important thing to note is that using the `nv_disable=1` boot argument prevents boot on High Sierra on Macs with discrete NVDA chips. Older versions of macOS are not affected (they just render things oddly, etc. but boot fine - which is desired).
